src/pages/index.js

```diff
@@ -76,24 +120,48 @@ function getCardElement(data) {
   const cardNameEl = cardElement.querySelector(".card__title");
   const cardImageEl = cardElement.querySelector(".card__image");
   const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const deleteButton = cardElement.querySelector(".card__delete-button");
 
-  cardLikeBtn.addEventListener("click", () => {
-    cardLikeBtn.classList.toggle("card__like-button_active");
-  });
+  cardNameEl.textContent = data.name;
+  cardImageEl.alt = data.name;
+  cardImageEl.src = data.link;
+
+  function handleLike(evt, id) {
+    const likeButton = evt.target;
+    const isLiked = likeButton.classList.contains("card__like-button_active");
+    api
+      .changeLikeStatus(id, isLiked)
+      .then(() => {
+        likeButton.classList.toggle("card__like-button_active");
+      })
+      .catch(console.error);
+  }
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button_active");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
 
   cardNameEl.textContent = data.name;
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
   cardImageEl.title = data.name;
 
-  cardLikeBtn.addEventListener("click", () => {
+  /*  cardLikeBtn.addEventListener("click", () => {
     cardLikeBtn.classList.toggle("card__like-button_liked");
   });
+ */
+  deleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
 
-  const deleteButton = cardElement.querySelector(".card__delete-button");
+  deleteModalCloseBtn.addEventListener("click", () => {
+    closeModal(deleteModal);
+  });
 
-  deleteButton.addEventListener("click", () => {
-    cardElement.remove();
+  deleteModalCancelBtn.addEventListener("click", () => {
```

Comment:
`deleteModalCloseBtn` and `deleteModalCancelBtn` should be handled **only once outside** `getCardElement`. It's not about cards. It's about the confirmation `modal`

Please, move this code into line 180

Reply:
I am not sure how to handle this. Don't they both need to be there? One so the modal closes when pressing cancel and the other for when you press the "X"?

Reply:
Everything is fine now. They are **outside** `getCardElement` now